hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java

```diff
@@ -1044,11 +1058,28 @@ public String toString() {
                 new NodeQueryCacheEndToEndConstructor(request));
     }
 
-    private void handleHazelcastInstanceAwareParams(Object... objects) {
-        for (Object object : objects) {
-            if (object instanceof HazelcastInstanceAware) {
-                ((HazelcastInstanceAware) object).setHazelcastInstance(getNodeEngine().getHazelcastInstance());
-            }
-        }
+    /**
+     * Handles injection of managed context into the provided object
+     * if any object implements a context-aware interface such as
+     * {@link com.hazelcast.core.HazelcastInstanceAware} or {@link com.hazelcast.spi.NodeAware}.
+     *
+     * @param o the objects which should have the contexts injected
+     */
+    private void initializeManagedContext(Object o) {
+        serializationService.getManagedContext().initialize(o);
+    }
+
+    /**
+     * Clones the object if the object does not implement {@link Immutable},
+     * otherwise returns the provided object
+     *
+     * @param object the object
+     * @param <T>    the type of the object
+     * @return the object clone or the provided object if it implements {@link Immutable}
+     */
+    private <T> T cloneIfNecessary(T object) {
```

Comment:
Alternative name: `cloneIfMutable()` (it's at least easier to spell and describes well what it does)